oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/LocalVocabularySelectField/LocalVocabularySelectField.jsx

```diff
@@ -130,36 +109,50 @@ export const LocalVocabularySelectField = ({
       processVocabularyItems(featuredOptions, showLeafsOnly, filterFunction),
     [featuredOptions, showLeafsOnly, filterFunction]
   );
-  const hasMultipleItems = multiple || fieldData.detail === "array";
 
   const handleChange = ({ data, formikProps }) => {
     if (hasMultipleItems) {
-      let vocabularyItems = allOptions.filter((o) =>
-        data.value.includes(o.value)
+      let vocabularyItems = serializedOptions.filter((o) =>
+        data.value.includes(o.id)
       );
       vocabularyItems = vocabularyItems.map((vocabularyItem) => {
-        return { ...vocabularyItem, id: vocabularyItem.value };
+        return { id: vocabularyItem.id };
       });
       formikProps.form.setFieldValue(fieldPath, [...vocabularyItems]);
     } else {
-      let vocabularyItem = allOptions.find((o) => o.value === data.value);
-      vocabularyItem = data.value
-        ? { ...vocabularyItem, id: vocabularyItem?.value }
-        : {};
+      let vocabularyItem = serializedOptions.find((o) => o.id === data.value);
+      vocabularyItem = data.value ? { id: vocabularyItem?.id } : {};
       formikProps.form.setFieldValue(fieldPath, vocabularyItem);
     }
   };
 
-  const { values, setFieldTouched } = useFormikContext();
-  const value = getIn(values, fieldPath, hasMultipleItems ? [] : {});
+  const customSearch = (options, searchQuery) =>
```

Comment:
could be just inlined in render as `search={(options, searchQuery) => search(options, searchQuery, "text")}`